components/script/dom/xmlhttprequest.rs

```diff
@@ -974,9 +974,15 @@ impl XMLHttpRequest {
                 *self.response.borrow_mut() = partial_response;
                 if !self.sync.get() {
                     if self.ready_state.get() == XMLHttpRequestState::HeadersReceived {
-                        self.change_ready_state(XMLHttpRequestState::Loading);
-                        return_if_fetch_was_terminated!();
+                        self.ready_state.set(XMLHttpRequestState::Loading);
                     }
+                    let global = self.global();
+                    let event = Event::new(global.r(),
+                           atom!("readystatechange"),
+                           EventBubbles::DoesNotBubble,
+                           EventCancelable::Cancelable);
```

Comment:
We also use that style of indentation in other places, and I think we should prefer it given it reduces the rightwards drift with longer expressions. Are you still blocking on this or are you ok with the current code @jdm?